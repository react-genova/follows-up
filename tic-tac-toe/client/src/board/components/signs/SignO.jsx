import React from 'react';
import PropTypes from 'prop-types';

const SignO = ({ ghost }) => (
    <svg width="100%" height="100%" viewBox="0 0 100 100">
        <ellipse cx="50" cy="50" rx="46" ry="46" fill="transparent" stroke={ghost ? 'lightgray' : 'yellow'} strokeWidth="8" />
    </svg>
);

SignO.propTypes = {
    ghost: PropTypes.bool,
};

SignO.defaultProps = {
    ghost: false,
};

export default SignO;
