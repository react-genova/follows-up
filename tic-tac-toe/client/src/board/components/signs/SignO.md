### On light background

```jsx
<div style={{ display:"flex", width:"100px", height:"50px", padding:"2px" }}>
    <SignO />
    <SignO ghost />
</div>
```

### On dark background

```jsx
<div style={{ display:"flex", width:"100px", height:"50px", padding:"2px", backgroundColor: "rgb(50,50,50)" }}>
    <SignO />
    <SignO ghost />
</div>
```

### A little bigger

```jsx
<div style={{ display:"flex", width:"200px", height:"100px", padding:"5px", backgroundColor: "rgb(50,50,50)" }}>
    <SignO />
    <SignO ghost />
</div>
```