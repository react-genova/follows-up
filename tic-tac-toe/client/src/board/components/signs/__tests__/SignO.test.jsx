import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';
import SignO from '../SignO';

describe('SignO component', () => {
    afterEach(cleanup);

    it('renders without crashing', () => {
        const { container } = render(<SignO />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders without crashing (ghost)', () => {
        const { container } = render(<SignO ghost />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
