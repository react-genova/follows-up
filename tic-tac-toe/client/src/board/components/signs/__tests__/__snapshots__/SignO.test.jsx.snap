// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignO component renders without crashing (ghost) 1`] = `
<svg
  class="sign-o"
  height="100%"
  viewBox="0 0 100 100"
  width="100%"
>
  <ellipse
    cx="50"
    cy="50"
    fill="transparent"
    rx="46"
    ry="46"
    stroke="lightgray"
    stroke-width="8"
  />
</svg>
`;

exports[`SignO component renders without crashing 1`] = `
<svg
  class="sign-o"
  height="100%"
  viewBox="0 0 100 100"
  width="100%"
>
  <ellipse
    cx="50"
    cy="50"
    fill="transparent"
    rx="46"
    ry="46"
    stroke="yellow"
    stroke-width="8"
  />
</svg>
`;
